### PR TITLE
Fix for Python 3.9/3.10: remove xml.etree.cElementTree

### PR DESCRIPTION
**Pull request by hugovk**
`xml.etree.cElementTree` has been deprecated since Python 3.3 and will be removed in Python 3.9 or Python 3.10 (https://github.com/python/cpython/pull/19921).

Since Python 3.3, `xml.etree.ElementTree` will use a fast implementation whenever available, so there's no need for the `try`/`except ImportError`.

* https://docs.python.org/3/library/xml.etree.elementtree.html

Also update the Trove classifiers to reflect Python 2.7 is no longer supported.

### CODE REVIEW

File: gensim/scripts/segment_wiki.py [hunk 1]
```diff
@@ -61,10 +61,7 @@
 import multiprocessing
 import re
 import sys
-try:
-    from xml.etree import cElementTree as ET
-except ImportError:
-    from xml.etree import ElementTree as ET
+from xml.etree import ElementTree as ET
```
Comment (piskvorky) [inline gensim/scripts/segment_wiki.py:64]: Not related to your PR, but I'm not a fan of such "shortcuts".

Can you please change this to use the full `ElementTree` or `xml.etree. ElementTree`?

Comment (hugovk) [reply gensim/scripts/segment_wiki.py:64]: Do you mean like this?

```python
import xml.etree.ElementTree as ET
...
elem = ET.fromstring(page_xml)
```

Or like this?

```python
import xml.etree.ElementTree
...
elem = ElementTree.fromstring(page_xml)
```

---

I'll note the former is used in the docs and quite common:

```python
import xml.etree.ElementTree as ET
tree = ET.parse('country_data.xml')
root = tree.getroot()
```

But happy to change to whichever you wish.

Comment (piskvorky) [reply gensim/scripts/segment_wiki.py:64]: `from xml.etree import ElementTree`. I don't want our code sprinkled with 2-letter acronym puzzles like `ET`.

They may be an obvious shorthand to the module's author, but not anybody else reading the code.

Comment (hugovk) [reply gensim/scripts/segment_wiki.py:64]: Updated!

**Pull request merged**